### PR TITLE
add support for rpm based runner installation

### PR DESCRIPTION
**Pull request by tourunen**
Add an option to install gitlab-runner as an rpm package on the runner
host. Initial testing shows that frequent transient errors encountered
with dockerized installation no longer occur, even if we are running the
same Docker-in-Docker mechanism for the jobs.

Changes
- select either rpm or docker based installation with
  'runner_deployment_type' (default=docker)
- option to set concurrency with 'runner_concurrency'

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -11,3 +11,5 @@ default_runner_config:
   RUNNER_NAME: "ubuntu"
   RUNNER_EXECUTOR: "docker"
   DOCKER_IMAGE: "ubuntu"
+
+runner_deployment_type: 'docker'
```
Comment (rlaurika) [inline defaults/main.yml]: Seems to be missing a newline.

File: tasks/main.yml [hunk 2]
```diff
@@ -89,11 +89,47 @@
...
+    path:  /srv/gitlab-runner/config/config.toml
+    line: 'concurrent = {{ runner_concurrency }}'
+    regexp: '^concurrent ='
+    state: present
+    insertbefore: 'BOF'
+    create: yes
   become: True
+  when: runner_concurrency is defined
 
 - include: gitlab_runner.yml
```
Comment (rlaurika) [inline tasks/main.yml]: Rename to gitlab_runner_docker.yml for consistency?

**Review by rlaurika — CHANGES_REQUESTED** [review]
Looks good! Since this is a rather substantial change, an update to the README.md file could be in order.

File: tasks/main.yml (hunk 2 above)
Comment (tourunen) [reply tasks/main.yml]: Yeah, I had that, but tried to minimize the change. I'll change that.

File: defaults/main.yml (hunk 1 above)
Comment (tourunen) [reply defaults/main.yml]: Indeed.

**Review by rlaurika — APPROVED** [review]

**Review by rlaurika — APPROVED** [review]
Looks good.

**Pull request merged**